_posts/2020-07-15-#19031.md

```diff
@@ -12,17 +12,22 @@ commit:
 
 ## Notes
 
-* [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is the
-  implementation of [BIP
-  155](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki), the
-  `addrv2` message.
+* [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is a proposed implementation of the [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) `addrv2` messages, a new p2p message format proposed in early 2019 by Wladimir J. van der Laan to gossip longer node addresses. This is required to support new-generation Onion addresses, I2P, and potentially other networks that have longer endpoint addresses than fit in the 128 bits of the current addr message.
 
-* The smaller PR that's currently being worked on, and should be reviewed
-  first, is [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360).
+* The proposed implementation in PR 19031 was turned in a roadmap and separated into a series of smaller PRs to ease review. The first one in the series, [PR 19351](https://github.com/bitcoin/bitcoin/pull/19351) "test: add an edge case test for CSubNet", has been merged. [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360) "net: improve encapsulation of CNetAddr" is the second PR in the series and currently under review.
 
-* Ultimately this will allow us to work with [Tor
-  v3](https://github.com/bitcoin/bitcoin/issues/18884), along with many other
-  different types of networks.
+* The `addr` message  relays connection information, including IP addresses, for peers on the Bitcoin network.  `addr` uses IPv6 addresses; IPv4 addresses are provided as IPv4-mapped IPv6 addresses.  More information [here](https://btcinformation.org/en/developer-reference#addr).
+
+* The BIP 155 `addrv2` messages would allow not only longer addresses, but also variable-length ones for future extensibility.
+
+* Bitcoin has Tor integration.  Relevant to PR 19031 is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address.  Tor v3 ([meta issue here](https://github.com/bitcoin/bitcoin/issues/18884)), AKA Next Gen Onion Services, has been out since 2017.  It requires 56-character addresses and therefore needs `addrv2`.  For more information on Tor v3, see [this page](https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions) on the Tor Bug Tracker and Wiki site.
+
+* PR 19031 is large and is still a WIP.  There's no reason to review all of it in detail.  Here is a suggested approach:
```

Comment:
19031 is not a WIP. It can be merged anytime if it gathers enough ACKs. If somebody has the resources - it makes perfect sense to review all of it too.

Reply:
I'll reword this.  My goal is to help people who don't have a lot of time and would otherwise be intimidated by the size of 19031.